packages/foam-vscode/src/core/services/markdown-link.test.ts

```diff
@@ -254,4 +254,140 @@ describe('MarkdownLink', () => {
       expect(edit.range).toEqual(link.range);
     });
   });
+
+  describe('convert wikilink to link', () => {
+    it('should generate default alias if no one', () => {
+      const wikilink = parser.parse(getRandomURI(), `[[wikilink]]`).links[0];
+      const wikilinkEdit = MarkdownLink.createUpdateLinkEdit(wikilink, {
+        type: 'link',
+      });
+      expect(wikilinkEdit.newText).toEqual(`[wikilink](wikilink)`);
+      expect(wikilinkEdit.range).toEqual(wikilink.range);
+
+      const wikilinkWithSection = parser.parse(
+        getRandomURI(),
+        `[[wikilink#section]]`
+      ).links[0];
+      const wikilinkWithSectionEdit = MarkdownLink.createUpdateLinkEdit(
+        wikilinkWithSection,
+        {
+          type: 'link',
+        }
+      );
+      expect(wikilinkWithSectionEdit.newText).toEqual(
+        `[wikilink#section](wikilink#section)`
+      );
+      expect(wikilinkWithSectionEdit.range).toEqual(wikilinkWithSection.range);
+    });
+
+    it('should use alias in the wikilik the if there has one', () => {
+      const wikilink = parser.parse(
+        getRandomURI(),
+        `[[wikilink#section|alias]]`
+      ).links[0];
+      const wikilinkEdit = MarkdownLink.createUpdateLinkEdit(wikilink, {
+        type: 'link',
+      });
+      expect(wikilinkEdit.newText).toEqual(`[alias](wikilink#section)`);
+      expect(wikilinkEdit.range).toEqual(wikilink.range);
+    });
+  });
+
+  describe('convert link to wikilink', () => {
+    it('should reorganize target, section, and alias in wikilink manner', () => {
+      const link = parser.parse(getRandomURI(), `[link](to/path.md)`).links[0];
+      const linkEdit = MarkdownLink.createUpdateLinkEdit(link, {
+        type: 'wikilink',
+      });
+      expect(linkEdit.newText).toEqual(`[[to/path.md|link]]`);
```

Comment:
See my comment above.
If we force passing the `target` this test would be straighfroward.
If we allow for a default target I would expect it to not have path and extension, so in this case it would be `[[path|link]]`